src/main/java/controllers/exercise03/StudentController.java

```diff
@@ -0,0 +1,108 @@
+package controllers.exercise03;
+
+import com.google.gson.reflect.TypeToken;
+import entities.exercise03.Student;
+import utils.FileUtils;
+
+import java.io.*;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+public class StudentController {
+    private static File file;
+
+    private static List<Student> students;
+
+    private static int COUNT;
+
+    private static Type type;
+
+
+    public StudentController(String filePath) {
+
+        students = new ArrayList<>();
+
+        file = new File(filePath);
+
+        loadFromFile();
+
+    }
+
+    public List<Student> getStudents() {
+        loadFromFile();
+
+        return students;
+    }
+
+    public void addStudent(Student student) {
+
+        Student obj = Student.StudentBuilder.newInstance()
+                .setId(COUNT++)
+                .setFirstName(student.getFirstName())
+                .setLastName(student.getLastName())
+                .setBirthday(student.getBirthday())
+                .setClassName(student.getClassName())
+                .setAddress(student.getAddress())
+                .build();
+
+        students.add(obj);
+        saveToFile();
+    }
+
+    public boolean deleteStudent(int id) {
+        boolean result = students.removeIf(o -> o.getId() == id);
+        saveToFile();
+        return result;
+    }
+
+    private void saveToFile() {
+        FileUtils.writeToJsonFile(file, students, type);
+    }
+
+    private void loadFromFile() {
+
+        type = new TypeToken<List<Student>>() {
+        }.getType();
+
+        FileUtils.readJsonFile(file, students, type);
+
+        COUNT = students.get(students.size() - 1).getId() + 1;
```

Comment:
Không rõ ý đồ lắm, COUNT đang nhận giá trị của `NEXT_AVAILABLE_ID`. nếu là count thì nó phải là 

```java
```
```suggestion
        COUNT = students.size();
```

Reply:
dạ count là giá trị NEXT_AVAILABLE_ID .. nếu lấy cái students.size nếu ta xóa bất kỳ 1 thằng nào trước đó rồi mới add mới vào thì id của thằng mới tạo ra sẽ trùng với thằng cuối cùng trong list.. nên e lấy ra id thằng cuối cùng sau e tăng lên 1 a..

Reply:
Em cứ tạo biến đếm tăng dần là xong, ví dụ em có danh sách students có id từ 1 đến 100, sau đó em xóa students có id từ 50 đến 100 thì cứ sinh viên tiếp theo sẽ là 101, không việc gì phải dùng lại id 51 cả.